senseme/lib/background_monitor.py

```diff
@@ -22,7 +24,12 @@ def __init__(self, interval=45, action=None):
 
     def _loop(self):
         while self.should_continue:
-            self.action()
+            try:
+                self.action()
+            except Exception as e:
+                # catch all exceptions to prevent loop from exiting
+                # when not intended
+                LOGGER.error("Background task Exception: %s" % str(e))
```

Comment:
If we change this to `LOGGER.exception` the `Exception as e` can be just `Exception` and the traceback would be logged automatically.

As to the overly broad exception, I agree, it is overly broad, but as generic as this code is, and for the purpose it is being used, I am okay with it.